src/include/execution/sql/index_iterator.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <memory>
+#include <vector>
+#include "catalog/catalog_defs.h"
+#include "execution/exec/execution_context.h"
+#include "execution/sql/projected_columns_iterator.h"
+#include "storage/storage_defs.h"
+
+namespace terrier::execution::sql {
+/**
+ * Allows iteration for indices from TPL.
+ */
+class IndexIterator {
```

Comment:
This class should be usable outside TPL. To help others understand how to use it, can you add sample code in the class comment?